### PR TITLE
Add The Topology ToolKit (TTK)

### PR DESCRIPTION
**Pull request by lhofmann**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -1,82 +0,0 @@
-# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
[the remark this answers is not in the document]
Comment (lhofmann) [reply recipes/example/meta.yaml]: Done. Thanks!

File: recipes/topologytoolkit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  skip: True  # [osx and py<37]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('libxi-devel') }}  # [linux]
```
[the remark this answers is not in the document]
Comment (lhofmann) [reply recipes/topologytoolkit/meta.yaml]: I was just following the [docs](https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl) and other packages, that depend on vtk, such as [smesh-feedstock](https://github.com/conda-forge/smesh-feedstock/blob/master/recipe/meta.yaml).
Anyways, I found that these dependencies are not actually needed and removed them.

**Comment by lhofmann** [comment]
@julien-tierny I have added you as maintainer, is that ok for you?
Also, please have a look at the meta data and feel free to edit!

**Comment by julien-tierny** [comment]
hi lutz,

all of this looks fantastic.

> @julien-tierny I have added you as maintainer, is that ok for you?
I have no experience with conda-forge so I'd appreciate if you could set yourself as a maintainer. thanks!

cheers
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

File: recipes/topologytoolkit/meta.yaml [hunk 3]
```diff
@@ -47,6 +47,7 @@ test:
 
 about:
   home: https://topology-tool-kit.github.io/
+  license: OTHER
```
Comment (julien-tierny) [inline recipes/topologytoolkit/meta.yaml]: BSD. btw, in principle, the license should be shipped somehow with the conda-forge binary distribution.

Comment (lhofmann) [reply recipes/topologytoolkit/meta.yaml]: What variant of BSD is this? 3-Clause BSD?
Btw, the file specified in `licence_file`, i.e. LICENSE from the ttk source, will be added to the package.

**Comment by lhofmann** [comment]
>> @julien-tierny I have added you as maintainer, is that ok for you?
> 
> I have no experience with conda-forge so I'd appreciate if you could set yourself as a maintainer. thanks! 

I have specified both of us as maintainers, and of course would be willing to do most of the maintaining (it's also my first conda package).
I just thought it would be a good idea to have more than one person with access to the package. What do you think?

**Comment by julien-tierny** [comment]
> What variant of BSD is this? 3-Clause BSD?
it's a variant of the original 4-clause BSD. 
if it's too exotic we can leave it at "other" for the moment.
> Btw, the file specified in `licence_file`, i.e. LICENSE from the ttk source, will be added to the package.
awesome.
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

**Comment by julien-tierny** [comment]
perfect, agreed 100%.
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

File: recipes/topologytoolkit/meta.yaml (hunk 3 above)
Comment (lhofmann) [reply recipes/topologytoolkit/meta.yaml]: > it's a variant of the original 4-clause BSD.
> if it's too exotic we can leave it at "other" for the moment.

@scopatz What would be the correct way to specify `license`/`license_family` in this case?

Comment (julien-tierny) [reply recipes/topologytoolkit/meta.yaml]: looks good to me.

**Comment by lhofmann** [comment]
@scopatz  Thanks for the help so far! Is there anything else, that needs to be revised?

**Pull request merged**